rpmlint/checks/BinariesCheck.py

```diff
@@ -277,26 +271,70 @@ def check_lto_section(self, pkg, path):
                     return
 
     def check_executable_stack(self, pkg, path):
-        if not path.endswith('.a'):
+        if not self.readelf_parser.is_archive:
             stack_headers = [h for h in self.readelf_parser.program_header_info.headers if h.name == 'GNU_STACK']
             if len(stack_headers) == 0:
                 self.output.add_info('E', pkg, 'missing-PT_GNU_STACK-section', path)
             elif 'E' in stack_headers[0].flags:
                 self.output.add_info('E', pkg, 'executable-stack', path)
 
+    def check_shared_library(self, pkg, path):
+        if not self.readelf_parser.is_shlib:
+            return
+        soname = self.readelf_parser.dynamic_section_info.soname
+        if not soname:
+            self.output.add_info('W', pkg, 'no-soname', path)
+        else:
+            if not self.validso_regex.search(soname):
+                self.output.add_info('E', pkg, 'invalid-soname', path, soname)
+            else:
+                (directory, base) = dir_base(path)
+                symlink = directory + soname
+                try:
+                    link = pkg.files()[symlink].linkto
```

Comment:
please describe this in some comment to have grasp on what goes on, also would be nice as separate function